src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -489,10 +1164,31 @@ func (s *commitLogSource) mergeSeries(
 		pooledBlock := blocksPool.Get()
 		pooledBlock.Reset(start, blockSize, enc.Discard())
 		if seriesBlocks == nil {
-			seriesBlocks = block.NewDatabaseSeriesBlocks(len(unmergedBlocks.encoders))
+			seriesBlocks = block.NewDatabaseSeriesBlocks(len(unmergedCommitlogBlocks.encoders))
 		}
 		seriesBlocks.AddBlock(pooledBlock)
 	}
+
+	if snapshotData.Blocks != nil {
+		allSnapshotBlocks := snapshotData.Blocks.AllBlocks()
+		for startNano, snapshotBlock := range snapshotData.Blocks.AllBlocks() {
+			if seriesBlocks == nil {
+				seriesBlocks = block.NewDatabaseSeriesBlocks(len(allSnapshotBlocks))
+			}
+			_, ok := seriesBlocks.BlockAt(startNano.ToTime())
+			if ok {
+				// Should never happen because we would have called
+				// Blocks.RemoveBlockAt() above.
+				iOpts := s.opts.CommitLogOptions().InstrumentOptions()
+				invariantLogger := instrument.EmitInvariantViolationAndGetLogger(iOpts)
+				invariantLogger.Errorf(
+					"Tried to merge block that should have been removed, blockStart: %d", startNano)
```

Comment:
nit: Lowercase the log entry for consistency.

Reply:
done